### DISCUSSION TITLE
Using UpdateResolver TS complains about "A function whose declared type is neither 'void' nor 'any' must return a value. ts(2355)"

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#1461] [Q&A] [answered]
I'm converting JS to TS and here I have this error:

`A function whose declared type is neither 'void' nor 'any' must return a value. ts(2355)` on line:

`playerCreate: (result, args, cache): UpdateResolver => {`

**Why?**

```ts
const updates = {
  Mutation: {
    playerCreate: (result, args, cache): UpdateResolver => {
      const playersQueries = cache
        .inspectFields("Query")
        .filter((x) => x.fieldName === "players");
      playersQueries.forEach(({ fieldName, arguments: variables }) =>
        cache.invalidate("Query", fieldName, variables)
      );
    },

    playerDelete: (result, args, cache, info): UpdateResolver => {
      // using result, args, cache here
    },
  },
};
```

I can see `Updateresolver` is declared like this:

```ts
export declare type UpdateResolver = (result: Data, args: Variables, cache: Cache, info: ResolveInfo) => void;
```

**Comment by JoviDeCroock** [comment, marked as answer]
You are saying that this function returns an `UpdateResolver`, while the type is for the function not the return-type

**Reply by frederikhors** [reply]
Oh wow. I'm new to TS. How can I write that?

**Reply by JoviDeCroock** [reply]
```
const playerCreate: UpdateResolver
```

I do think this should already happen implicitly though

**Reply by frederikhors** [reply]
I tried, it says: `'UpdateResolver' only refers to a type, but is being used as a value here. ts(2693)`.

**Reply by frederikhors** [reply]
I opened https://stackoverflow.com/questions/66693524/how-can-i-correctly-type-functions-in-object-using-typescript-here.

**Comment by frederikhors** [comment]
I created this type:

```ts
type Updates = {
  Mutation: {
    playerCreate: UpdateResolver;
    playerDelete: UpdateResolver;
  };
};
```

and I'm using it like this:

```ts
const updates: Updates = {
```

Is there a better way?

**Reply by frederikhors** [reply]
@JoviDeCroock what do you think about this?
